lnwire/onion_error.go

```diff
@@ -0,0 +1,879 @@
+	CodeInvalidOnionHmac                       = FlagBadOnion | FlagPerm | 5
+	CodeInvalidOnionKey                        = FlagBadOnion | FlagPerm | 6
+	CodeTemporaryChannelFailure                = FlagUpdate | 7
+	CodePermanentChannelFailure                = FlagPerm | 8
+	CodeRequiredChannelFeatureMissing          = FlagPerm | 9
+	CodeUnknownNextPeer                        = FlagPerm | 10
+	CodeAmountBelowMinimum                     = FlagUpdate | 11
+	CodeFeeInsufficient                        = FlagUpdate | 12
+	CodeIncorrectCltvExpiry                    = FlagUpdate | 13
+	CodeExpiryTooSoon                          = FlagUpdate | 14
+	CodeChannelDisabled                        = FlagUpdate | 20
+	CodeUnknownPaymentHash                     = FlagPerm | 15
+	CodeIncorrectPaymentAmount                 = FlagPerm | 16
+	CodeFinalExpiryTooSoon                     = 17
+	CodeFinalIncorrectCltvExpiry               = 18
+	CodeFinalIncorrectHtlcAmount               = 19
+)
+
+// String returns string representation of the failure code.
+func (c FailCode) String() string {
+	switch c {
+	case CodeInvalidRealm:
+		return "InvalidRealm"
+
+	case CodeTemporaryNodeFailure:
+		return "TemporaryNodeFailure"
+
+	case CodePermanentNodeFailure:
+		return "PermanentNodeFailure"
+
+	case CodeRequiredNodeFeatureMissing:
+		return "RequiredNodeFeatureMissing"
+
+	case CodeInvalidOnionVersion:
+		return "InvalidOnionVersion"
+
+	case CodeInvalidOnionHmac:
+		return "InvalidOnionHmac"
+
+	case CodeInvalidOnionKey:
+		return "InvalidOnionKey"
+
+	case CodeTemporaryChannelFailure:
+		return "TemporaryChannelFailure"
+
+	case CodePermanentChannelFailure:
+		return "PermanentChannelFailure"
+
+	case CodeRequiredChannelFeatureMissing:
+		return "RequiredChannelFeatureMissing"
+
+	case CodeUnknownNextPeer:
+		return "UnknownNextPeer"
+
+	case CodeAmountBelowMinimum:
+		return "AmountBelowMinimum"
+
+	case CodeFeeInsufficient:
+		return "FeeInsufficient"
+
+	case CodeIncorrectCltvExpiry:
+		return "IncorrectCltvExpiry"
+
+	case CodeExpiryTooSoon:
+		return "ExpiryTooSoon"
+
+	case CodeChannelDisabled:
+		return "ChannelDisabled"
+
+	case CodeUnknownPaymentHash:
+		return "UnknownPaymentHash"
+
+	case CodeIncorrectPaymentAmount:
+		return "IncorrectPaymentAmount"
+
+	case CodeFinalExpiryTooSoon:
+		return "FinalExpiryTooSoon"
+
+	case CodeFinalIncorrectCltvExpiry:
+		return "FinalIncorrectCltvExpiry"
+
+	case CodeFinalIncorrectHtlcAmount:
+		return "FinalIncorrectHtlcAmount"
+
+	default:
+		return "<unknown>"
+	}
+}
+
+// FailInvalidRealm is returned if the realm byte is unknown.
+// NOTE: might be returned by any node.
+type FailInvalidRealm struct{}
+
+// Code returns the failure unique code.
+// NOTE: Part of the Failure interface.
+func (f FailInvalidRealm) Code() FailCode {
+	return CodeInvalidRealm
+}
+
+// FailTemporaryNodeFailure is returned if an otherwise unspecified transient
+// error occurs for the entire node.
+// NOTE: might be returned by any node.
```

Comment:
I think that keeping it as a distinct note, might reduce time for programmer to better understand what expect from this error, without reading full description.